src/adapter/graphql/src/root.rs

```diff
@@ -78,6 +79,28 @@ impl Query {
     async fn molecule(&self) -> Molecule {
         Molecule
     }
+
+    // Federation
+
+    #[graphql(entity)]
+    #[tracing::instrument(level = "info", name = Query_find_account_by_id, skip_all, fields(%account_id))]
+    async fn find_account_by_id(
+        &self,
+        ctx: &Context<'_>,
+        account_id: AccountID<'_>,
+    ) -> Result<Option<Account>> {
+        Accounts.by_id(ctx, account_id).await
+    }
+
+    #[graphql(entity)]
+    #[tracing::instrument(level = "info", name = Query_find_dataset_by_id, skip_all, fields(%dataset_id))]
+    async fn find_dataset_by_id(
+        &self,
+        ctx: &Context<'_>,
+        dataset_id: DatasetID<'_>,
+    ) -> Result<Option<Dataset>> {
+        Datasets.by_id(ctx, dataset_id).await
+    }
 }
```

Comment:
Why does this have to be at the root level?

Reply:
Yes, unfortunately, that's the only way

However, these methods are not explicitly shown in the GQL schema

Reply:
Updated the comment